### PR TITLE
Nb 146 keyboard overlaps inputs make it responsive

### PR DESCRIPTION
**Pull request by nemanja255**
# Description

Solved the issue of iOS devices keyboard overlapping form input fields. 

# How Has This Been Tested?

- [x] Tested all of the changes on both iOS and Android devices.
- [x] Tested all occurrences of input fields and modals with the current changes. Did not manage to break design with new changes.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

# New Components

- KeyboardAvoidingComponent

# Modified Components

- NblocksModalComponent
- LoginScreen
- BrandExpoScreen

### CODE REVIEW

File: src/components/shared/KeyboardAvoidingComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    return (
+      <ScrollView
+        style={{ flex: 1 }}
+        overScrollMode={config.scrollMode}
+        bounces={config.scrollBounces}
+        contentContainerStyle={config.contentContainerStyle}
+        showsVerticalScrollIndicator={config.showScrollIndicator}
+        keyboardDismissMode={config.keyboardDismissMode}
+      >
+        {useBehavior ? (
```
Comment (orr3) [inline src/components/shared/KeyboardAvoidingComponent.tsx:63]: This block from #66 to #78 is identical with #84 to #96.
Can you put this into a separate function that can be called instead?

See how I use `renderExpanedContent` in `UserListComponent` for inspo.

File: src/screens/Auth/LoginScreen.tsx [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
   return (
-    <DefaultPaddingComponent style={{flex: 1}}>
-      <LoginComponent />
-    </DefaultPaddingComponent>
+    <ScrollView
+      style={{ flex: 1 }}
+      contentContainerStyle={{ flexGrow: 1 }}
+      keyboardDismissMode={Platform.OS === "ios" ? "interactive" : "on-drag"}
+    >
+      <DefaultPaddingComponent style={{ flex: 1 }}>
```
Comment (orr3) [inline src/screens/Auth/LoginScreen.tsx:12]: Is (flex: 1) required to remain on DefaultPaddingComponent now when there's a parent view (scrollview) with flex:1?

**Review by orr3 — CHANGES_REQUESTED** [review]
Small feedback

File: src/screens/Auth/LoginScreen.tsx (hunk 2 above)
Comment (nemanja255) [reply src/screens/Auth/LoginScreen.tsx:12]: Yeah, react native is based on flex. The nested DefaultPaddingComponent grows to fit the wrapper container which in turn grows to fit its parent... The only walk-around would be not using the DefaultPaddingComponent and set the padding on the ScrollView.

**Pull request merged**